src/test/testFixtureUtils.tsx

```diff
@@ -215,34 +227,43 @@ export async function loadFixtureData(testName: string): Promise<UIStore> {
   return store;
 }
 
-export function sendLoadedRegionsToMockEnvironment(value: loadedRegions): void {
-  window.mockEnvironment?.sendSocketMessage(
-    JSON.stringify({
-      method: "Session.loadedRegions",
-      params: value,
-      sessionId: DEFAULT_SESSION_ID,
-    })
-  );
-}
-
-export function sendMessageToMockEnvironment(value: Message): void {
-  window.mockEnvironment?.sendSocketMessage(
-    JSON.stringify({
-      method: "Console.newMessage",
-      params: {
-        message: value,
-        sessionId: DEFAULT_SESSION_ID,
-      },
-    })
-  );
-}
+type Tuple = LoadedRegionsTuple | MessageTuple | SourceTuple;
 
-export function sendSourceToMockEnvironment(value: newSource): void {
-  window.mockEnvironment?.sendSocketMessage(
-    JSON.stringify({
-      method: "Debugger.newSource",
-      params: value,
-      sessionId: DEFAULT_SESSION_ID,
-    })
-  );
+export function sendValuesToMockEnvironment(...values: Array<Tuple>): void {
+  values.forEach(([method, value]) => {
+    switch (method) {
+      case "Session.loadedRegions":
+        window.mockEnvironment?.sendSocketMessage(
+          JSON.stringify({
```

Comment:
:speech_balloon: This is all test util code and not really a big deal to keep DRY. At the same time, I'm looking at this and feeling like it could be turned into either a lookup table, or something that does formatting first and then a single call to `sendSocketMessage()` at the end?

Reply:
This is a draft PR and that's a nit pick 😉

Reply:
A more serious answer: I had hoped we'd be able to eventually type the values we're passing to `sendSocketMessage` message too (and going too DRY like you're suggesting might interfere with that). That being said, I'm not sure if it's possible (if we even have that typing information available) so....